src/VRCWrapper.ts

```diff
@@ -0,0 +1,214 @@
+
+import cookiesHandler from './VRCCookie';
+import { Color as C } from './colors';
+import { CookiesExpired, CookiesNotFound, CookiesReadError, CookiesUser404, CookiesWriteError, RequestError, TOTPRequired } from './errors';
+import { AuthApi } from './requests/AuthApi';
+import { GroupsApi } from './requests/GroupsApi';
+import { UsersApi } from './requests/UsersApi';
+import { ApiPaths } from './types/ApiPaths';
+
+/**
+ * This class is used to authenticate the user and get the current user information.
+ * @export
+ * @class VRCWrapper
+ */
+export class VRCWrapper {
+
+  username: string;
+  password: string;
+  public instanceCookie: cookiesHandler;
+  isAuthentificated: boolean;
+  static baseDomain: string = 'api.vrchat.cloud';
+  public static ApiBaseUrl: string = "https://api.vrchat.cloud/api/1"
+  headerAgent: string = process.env.USER_AGENT || "ExampleApp/1.0.0 Email@example.com";
+  basePath: string = ApiPaths.api.base.path;
+
+  cookiesLoaded = false;
+
+  authApi: AuthApi;
+  userApi: UsersApi;
+  groupApi: GroupsApi;
+
+
+
+  constructor(username: string, password: string) {
+    this.username = username;
+    this.password = password;
+    this.isAuthentificated = false;
+    this.instanceCookie = new cookiesHandler(this.username)
+    this.instanceCookie = new cookiesHandler(this.username);
+    this.authApi = new AuthApi(this);
+    this.userApi = new UsersApi(this);
+    this.groupApi = new GroupsApi(this);
+  }
+
+  async authenticate() {
+
+
+    if (process.env.USE_COOKIES && process.env.USE_COOKIES === 'true') {
+      try {
+        await this.instanceCookie.cookiesExist();
+        await this.instanceCookie.loadCookies();
+        this.cookiesLoaded = true;
+      } catch (error) {
+        this.cookiesLoaded = false;
+        if (error instanceof CookiesExpired) {
+          console.error(`${C.yellow} [WARNING]${C.r} - Cookies are expired, a new session will be created.`);
+        }
+        if (error instanceof CookiesNotFound || error instanceof CookiesUser404) {
+          this.cookiesLoaded = false;
+        } else if (error instanceof Error) {
+          console.error("Error loading cookies: ", error.message);
+        } else {
+          console.error("Unknown error loading cookies!");
+        }
```

Comment:
All these errors should be in the constructors of the errors so they log these when they are Throw.

Reply:
The main VRCWrapper will be reworked a bit maybe tomorrow, I already tweaked it a bit tonight, it's getting more and more clean now.